### DISCUSSION TITLE
protobuf port

### DISCUSSION THREAD

**Discussion by stbjones** [zopencommunity/meta#361] [General]
Hi.

I see that a port of google protobuf was started, although the last person to commit is no longer found in blue pages.  Does anyone have an idea of the status of that port?  In building a new release of Apache Spark for our product, IBM Z Platform for Apache Spark, I found that the Spark developers now include protobuf version 3.21.12, which has a check of os.name and gives an error, 
`Error resolving artifact: com.google.protobuf:protoc:3.21.12: unknown os.name: z/OS -> [Help 1]`

Any thoughts?

**Comment by IgorTodorovskiIBM** [comment]
This port is out of date. I suggest you reach out to @ccw-1 . I believe his team ported protobuf as a dependency for the Go compiler, you may be able to leverage that. Ultimately we should get those changes open sourced.

**Comment by IgorTodorovskiIBM** [comment]
Hi @stbjones , we have a build of the latest version of protobuf now available here: https://github.com/ZOSOpenTools/protobufport/releases

You can follow these instructions for Go support: https://chromium.googlesource.com/external/github.com/golang/protobuf/+/e09c5db296004fbe3f74490e84dcd62c3c5ddb1b/README.md

**Reply by stbjones** [reply]
Hi @IgorTodorovskiIBM .  For protobuf, and Spark in general, we assume everything has ASCII encoding.  Is that how this level is built?  Any suggestions on environment vars, etc. to run with to help ensure success?

Thanks!
